core/src/main/scala/org/bitcoins/core/wallet/builder/RawTxFinalizer.scala

```diff
@@ -280,6 +280,67 @@ object StandardNonInteractiveFinalizer {
   }
 }
 
+/** A finalizer which adds a change output, performs sanity checks,
+  * shuffles inputs and outputs, and adds non-signature witness data.
+  * This is the standard non-interactive finalizer within the Bitcoin-S wallet.
+  */
+case class ShufflingNonInteractiveFinalizer(
+    inputInfos: Vector[InputInfo],
+    feeRate: FeeUnit,
+    changeSPK: ScriptPubKey)
+    extends RawTxFinalizer {
+
+  override def buildTx(txBuilderResult: RawTxBuilderResult)(implicit
+      ec: ExecutionContext): Future[Transaction] = {
+    val addChange = ChangeFinalizer(inputInfos, feeRate, changeSPK)
+
+    val sanityCheck = SanityCheckFinalizer(
+      inputInfos = inputInfos,
+      expectedOutputSPKs = txBuilderResult.outputs.map(_.scriptPubKey),
+      expectedFeeRate = feeRate,
+      changeSPKs = Vector(changeSPK))
+
+    val addWitnessData = AddWitnessDataFinalizer(inputInfos)
+
+    addChange
+      .andThen(sanityCheck)
+      .andThen(addWitnessData)
+      .andThen(ShuffleFinalizer)
+      .buildTx(txBuilderResult)
```

Comment:
`addWitnessData` *MUST* happen last otherwise its effects are dropped (as per its scaladoc).
`sanityCheck` *SHOULD* happen second to last so that we can sanity check the inputs and outputs post-shuffling.
Thus, this should be
```
addChange
    .andThen(ShuffleFinalizer)
    .andThen(sanityCheck)
    .andThen(addWitnessData)
    .buildTx(txBuilderResult)
```